qt/widgets/plotting/src/Mpl/ExternalPlotter.cpp

```diff
@@ -103,24 +102,35 @@ boost::optional<Python::Object> workbenchPlot(QStringList const &workspaceNames,
 } // namespace
 
 namespace MantidQt {
-namespace CustomInterfaces {
+namespace Widgets {
+namespace MplCpp {
 
 ExternalPlotter::ExternalPlotter() {}
 
 ExternalPlotter::~ExternalPlotter() {}
 
+void ExternalPlotter::plotSpectra(std::string const &workspaceName, std::string const &workspaceIndices,
+                                  bool errorBars) {
+  return plotSpectra(workspaceName, workspaceIndices, errorBars, boost::none);
+}
+
 /**
  * Produces an external plot of workspace spectra
  *
  * @param workspaceName The name of the workspace to plot
  * @param workspaceIndices The indices within the workspace to plot (e.g.
  * '0-2,5,7-10')
  */
-void ExternalPlotter::plotSpectra(std::string const &workspaceName, std::string const &workspaceIndices,
-                                  bool errorBars) {
+void ExternalPlotter::plotSpectra(std::string const &workspaceName, std::string const &workspaceIndices, bool errorBars,
+                                  boost::optional<QHash<QString, QVariant>> kwargs) {
```

Comment:
Can we do `boost::optional<QHash<QString, QVariant>> kwargs = boost::none`, and then we won't need the if statement below

Reply:
Would it be safe to remove the if statement anyway since if no `kwargs` are being passed the other plotSpectra function should be called which in turns calls plotSpectra with `kwargs = boost::none`?

Reply:
Ah yes, it would be safe just to remove `kwargs` without doing `kwargs = boost::none`. However you would need to pass in `boost::none` from everywhere in indirect that uses `plotSpectra`.

It would be nice just to have one `plotSpectra` function, can you show me the problem you were having with the test?